cogs/hosts.py

```diff
@@ -129,78 +130,82 @@ async def delhost(self, ctx, user: discord.Option(discord.Member, description="d
             not_nobot_role_slash()
         ]
     )
-    async def temphost(self, ctx):
+    async def temphost(self, ctx, user: discord.Option(discord.Member, description="discord user")):
+     
+        #check if host already
+        hosts_role = discord.utils.get(ctx.guild.roles, name="Hosts")
 
-        #check if vip first
-        vip_role = discord.utils.get(ctx.guild.roles, name="VIP")
-
-        if vip_role not in ctx.author.roles:
-            await ctx.respond(f"{warning} You're not a vip, only vip's may get temp-hosting powers for gamedays.")
+        if hosts_role in ctx.author.roles:
+            await ctx.respond(f"{warning} They're a host already. Go away.")
             return
+        
+        #check if temp-host already
+        temphosts_role = discord.utils.get(ctx.guild.roles, name="Temp-Host")
 
+        if temphosts_role in ctx.author.roles:
+            await ctx.respond(f"{warning} They're a temp-host already. Go away.")
+            return
 
-        #
-        # check if user has forum account
-        #
-        await ctx.respond(f"{loading} Adding you to the temp-hosts forum group...")
+        #do site things
+        await ctx.respond(f"{loading} Adding {user.name} to hosts group...")
         time.sleep(2)
         database = databases.Database(global_config.databases.tf2maps_site)
         await database.connect()
         
         #get their forum ID
         query = "SELECT user_id FROM xf_user_connected_account WHERE provider_key = :field_value"
-        values = {"field_value": ctx.author.id}
+        values = {"field_value": user.id}
         result = await database.fetch_one(query=query, values=values)
 
         #has connected discord account check
         if not result:
-            await ctx.edit(content=f"{error} You don't have a Discord User ID # set in your TF2Maps.net profile.\nSee [this help page](<http://bot.tf2maps.net/faq.php>) on how to get started.")
+            await ctx.edit(content=f"{error} {user.name} doesn't have a Discord User ID # set in their TF2Maps.net profile.\nSee [this help page](<http://bot.tf2maps.net/faq.php>) on how to get started.")
             return
+
+        user_id = result[0]
+        user_secondary_groups = await self.get_user_roles(user_id)
+
+        if 43 in user_secondary_groups:
+            await ctx.edit(content=f"{warning} {user.name} already has the temp-host group on the forum! Checking discord...")
+            time.sleep(3)
+        else:
+            message = await ctx.edit(content=f"{loading} Adding {user.name} to temp-host group on the forums...")
+            time.sleep(2)
```

Comment:
use `await asyncio.sleep(2)` instead

time.sleep will block the async thread and cause the bot to completely freeze. do this for any sleep you need to put in. also why do we even need to sleep here?